src/PKSim.Core/Services/CKDDiseaseStateImplementation.cs

```diff
@@ -53,50 +60,68 @@ public void ApplyTo(Individual individual)
          var gastricEmptyingTime = stomach.Parameter(GASTRIC_EMPTYING_TIME);
 
-         var bsaInm2 = bsa.ConvertToUnit(bsa.Value, CoreConstants.Units.m2);
-         var GFR_0 = GFR_spec.Value * kidneyVolume.Value * 1.73 / bsaInm2;
-         if (GFR_0 < targetGFR.Value)
+
+         var GFR_0 = getTargetGFRValue(GFR_spec.Value * kidneyVolume.Value / bsa.Value);
+         var targetGFRValue = getTargetGFRValue(targetGFR.Value);
+         if (GFR_0 < targetGFRValue)
             throw new OSPSuiteException("ERROR");
 
          //Adjust kidney volume and update fat accordingly
          var healthyKidneyVolume = kidneyVolume.Value;
-         kidneyVolume.Value = getKidneyVolumeFactor(targetGFR.Value) / getKidneyVolumeFactor(GFR_0) * healthyKidneyVolume;
-         fatVolume.Value = fatVolume.Value + healthyKidneyVolume - kidneyVolume.Value;
+         updateParameter(kidneyVolume, getKidneyVolumeFactor(targetGFRValue) / getKidneyVolumeFactor(GFR_0) * healthyKidneyVolume);
+         updateParameter(fatVolume, fatVolume.Value + healthyKidneyVolume - kidneyVolume.Value);
 
-         //Adjust rena blood flow
-         kidneySpecificBloodFlowRate.Value = getRenalBloodFlowFactor(targetGFR.Value) / getRenalBloodFlowFactor(GFR_0) * kidneySpecificBloodFlowRate.Value;
+         //Adjust renal blood flow spec
+         updateParameter(kidneySpecificBloodFlowRate, getRenalBloodFlowFactor(targetGFRValue) / getRenalBloodFlowFactor(GFR_0) * kidneySpecificBloodFlowRate.Value);
 
          //Correct specific GFR
-         GFR_spec.Value = GFR_spec.Value * targetGFR.Value / GFR_0 * healthyKidneyVolume / kidneyVolume.Value;
+         updateParameter(GFR_spec, GFR_spec.Value * targetGFRValue / GFR_0 * healthyKidneyVolume / kidneyVolume.Value);
 
-         var ckdStage = getCKDStage(targetGFR);
-         var (plasmaProteinScaleFactor, gastricEmptyingTimeFactor, smallIntestinalTransitTimeFactor) = getCategorialFactors(ckdStage);
+     
+         var (plasmaProteinScaleFactor, gastricEmptyingTimeFactor, smallIntestinalTransitTimeFactor) = getCategorialFactors(targetGFRValue);
 
          //Categorial Parameters
-         plasmaProteinScaleFactorParameter.Value = plasmaProteinScaleFactor;
-         gastricEmptyingTime.Value *= gastricEmptyingTimeFactor;
-         smallIntestinalTransitTime.Value *= smallIntestinalTransitTimeFactor;
+         updateParameter(plasmaProteinScaleFactorParameter, plasmaProteinScaleFactor);
+         updateParameter(gastricEmptyingTime, gastricEmptyingTime.Value * gastricEmptyingTimeFactor);
+         updateParameter(smallIntestinalTransitTime, smallIntestinalTransitTime.Value * smallIntestinalTransitTimeFactor);
 
          //Special case for Hematocrit
-         hct.Value *= getHematocritFactor(targetGFR, individual.OriginData.Gender);
+         updateParameter(hct, hct.Value * getHematocritFactor(targetGFRValue, individual.OriginData.Gender));
+      }
+
+      private void updateParameter(IParameter parameter, double value)
```

Comment:
@Yuri05 I think this is enough but I might be mistaken
For distributed parmaeter, I used the scale method. This will keep the percentile 

For Explicit formula (GFR_Spec), I create a constant
Othewise, override value and deafult